### DISCUSSION TITLE
Vercel Dev Local Development - Debug / Output / Hot reload

### DISCUSSION THREAD

**Discussion by lucashfreitas** [vercel/vercel#5527] [General] [answered]
Hello everybody,

I have been using netlify functions for a while and just switched to `vercel cli` recently. I am using the latest version `21.0.1` and I've got a few questions:

1 - My serverless function returns 500 but `vercel dev` doesn't output any error or stacktrace in the console. Is that normal or might be a problem with my local environment setup?

2 - There is any news on the debug feature? Currently, it's possible to debug netlify functions but not too sure if it is also available on vercel dev?

3 - There is any way to reload functions whenever changes are made to them?

Thaaanks :)

**Comment by TooTallNate** [comment, marked as answer]
Hello there @lucashfreitas! Happy to have you here. To answer your questions:

1. `vercel dev` should match production as closely as possible, so I'd expect it to return a 500 as well. If you have a small reproduction of a discrepancy, I'd be happy to take a look.
1. How do you mean "debug"? There is a `--debug` flag but it just prints some extra logging to the console.
1. What language are your serverless functions written in? If Node.js or Go, then each HTTP request to the function is a fresh new process and is essentially a "cold-start", so you're guaranteed to be using the latest version of the source files. You could verify that by checking i.e. `process.pid` from within your serverless function code.

Hope that helps!

**Reply by jeantil** [reply]
>How do you mean "debug"? There is a --debug flag but it just prints some extra logging to the console

I think he means being able to use a debugger with breakpoints which was reported to be in the works in https://github.com/vercel/vercel/issues/3849#issuecomment-620157834

**Comment by lucashfreitas** [comment]
hey @TooTallNate , sorry for the late reply:

1 - I mean when an exception is thrown the console didn't output the error by default.
2 - I mean like @jeantil said, debug with breakpoints
3 - Cool, didn't know that we always use the latest code/version.

Thaanks!
